src/node/history.h

```diff
@@ -381,6 +381,20 @@ namespace ccf
     std::optional<ResultCallbackHandler> on_result;
     std::optional<ResponseCallbackHandler> on_response;
 
+    void discard_pending(kv::Version v)
+    {
+      auto* p = pending_inserts.get_head();
```

Comment:
pending_inserts is not thread safe it can only be accessed under a lock
please have a look at execute_pending.

Also while you are here could you please rename execute pending to flush pending. I did not give it a great name :S